imports/plugins/core/catalog/server/no-meteor/startup.js

```diff
@@ -32,6 +33,9 @@ async function hashRelatedProduct(media, collections) {
 export default function startup(context) {
   const { appEvents, collections } = context;
 
+  collectionIndex(collections.Catalog, { "product.productId": 1 }); // Used to update products
+  collectionIndex(collections.Catalog, { "product.variants.options._id": 1 }); // Used while publishing products
+
```

Comment:
We had a team conversation in Slack where it was proposed to combine the id's and slugs from all the levels into a single field that could be indexed. We had a technical MongoDB review from the Atlas team that strongly recommended that we reduce the number of indexes. It was the highest priority recommendation from the review.

Could we discuss and determine a plan for creating the single index field before merging this?

cc @spencern @mikemurray

Reply:
I reviewed indexes yesterday while doing #5090. Although my PR does not add or remove any indexes yet, I did note many indexes that don't seem to be necessary in my view. These two do seem to be necessary. I think we should add these and then remove all of the blatantly useless ones and re-evaluate after that.

Reply:
If adding this index solves a problem that we're having now without having to rewrite code (single index field), I'd like to get this merged and revisit optimizations to our queries tomorrow, from there we can determine and remove unnecessary indexes.

Ticean do you feel we should hold up this PR until we make a decision about indexing better? or can we create a follow up issue for someone to work this week?

Reply:
@spencern depends on what you're thinking when you say "a problem that we're having now". We have an implementation in progress but indexes can be added to that project's database manually, which is what the team is doing for now. I don't see why there's a rush to make a merge to Reaction core where we have the potential to churn indexes on the open source users.

The reason I asked that we hold is that the design of this particular merge is that these two indexes don't provide full coverage over the app queries. Scans from the Catalog collections were big part of why we had so many alerts from Atlas on the project.

Reply:
So @ticean, @spencern @aldeed where are we with this? Are we closing this, are we doing anything to move this forward?